core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -374,4 +374,21 @@ TableScanContext reportWith(MetricsReporter reporter) {
         fromSnapshotInclusive,
         reporter);
   }
+
+  TableScanContext withColStats(boolean stats) {
```

Comment:
I think this is no different compared to `shouldReturnColumnStats` above, which is already used by scans.
If we switch to `scan$includeColumnStats`, this won't be needed.

Reply:
I thought I'd checked for an existing method and didn't see one. My bad here.